codalab/worker/rest_client.py

```diff
@@ -160,7 +163,21 @@ def _upload_with_chunked_encoding(
             conn.send(b'0\r\n\r\n')
 
             # Read the response.
-            response = conn.getresponse()
+            logging.debug("About to read the response... url: %s", url)
+
+            # Sometimes, it may take a while for the server to process
+            # the data and send the response. In this case, we want to
+            # periodically keep sending empty bytes so that the
+            # connection doesn't drop before the response is available.
+            got_response = False
+            while not got_response:
+                try:
+                    response = conn.getresponse()
+                    got_response = True
+                except socket.timeout:
+                    logging.debug("Socket timeout, retrying")
```

Comment:
We should probably log the number of retries to get, so we have a sense of how many retries in general we need to upload large bundles. (If we fail after a set number of tries / dynamic number of tries, we should log that too)

Reply:
Good idea, I just logged the URL for now, so that we can distinguish the socket retries and count the number of logs per URL if need be.